Update tcia prostatex model notebook

kirbyju opened the pull request:
### Description
Fixed dependency issues with itkwidgets

kirbyju committed:
Fixed dependency issues with itkwidgets

kirbyju committed:
Revert "Fixed dependency issues with itkwidgets"

This reverts commit 00f3362af5df6e279e5978b6117517c34adf93cd.

kirbyju committed:
Updated tcia_prostatex_prostate_mri_anatomy_model.ipynb

Fixed some dependency issues related to itkwidgets that were preventing the notebook from running.

kirbyju committed:
Merge branch 'main' into update-tcia-prostatex-model-notebook

thewtex commented on model_zoo/TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb:
```diff
@@ -116,17 +116,21 @@
+    "\n",
     "# This is the installation required for itkWidgets.\n",
-    "!python -m pip install --upgrade --pre -q \"itkwidgets[all]==1.0a23\" imjoy_elfinder"
+    "!python -m pip install --upgrade --pre -q \"itkwidgets[all]\" imjoy_elfinder\n",
```
@kirbyju thanks for addressing this!

To clarify, we want to specify the pre-release version of itkwidgets:

```
itkwidgets[all]==1.0a53
```

but avoid/remove the `--pre` flag because it can bring in unstable transitive dependencies like `zarr==3.0.0a1`.

kirbyju commented on model_zoo/TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb:
```diff
@@ -116,17 +116,21 @@
     "\n",
     "# Install the dependency manually to avoid installing opencv-python.\n",
-    "!python -m pip install -q plotly bs4 ipywidgets unidecode jsonschema\n",
+    "!python -m pip install -q plotly bs4 unidecode\n",
```
ipywidgets is no longer required by tcia_utils and jsonschema gets installed by pydicom-seg

kirbyju committed:
Update TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb

Updated with the guidance provided in https://github.com/Project-MONAI/tutorials/pull/1852.

KumoLiu commented:
Looks the notebook still can not pass the premerge.
https://github.com/Project-MONAI/tutorials/actions/runs/11241293887/job/31252532094?pr=1852

kirbyju committed:
Justin Kirby <2379527+kirbyju@users.noreply.github.com>
DCO Remediation Commit for Justin Kirby <2379527+kirbyju@users.noreply.github.com>

I, Justin Kirby <2379527+kirbyju@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 00f3362af5df6e279e5978b6117517c34adf93cd
I, Justin Kirby <2379527+kirbyju@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 6e36c647472b96e1bf01ead6a2f37550e9b4176a
I, Justin Kirby <2379527+kirbyju@users.noreply.github.com>, hereby add my Signed-off-by to this commit: f3d172ea744f305dbe0937a71684f7c92365f208
I, Justin Kirby <2379527+kirbyju@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 6fc1435c5431d5a84237cfb0f2db4f759dea1ea6

Signed-off-by: Justin Kirby <2379527+kirbyju@users.noreply.github.com>

kirbyju commented:
> Looks the notebook still can not pass the premerge. https://github.com/Project-MONAI/tutorials/actions/runs/11241293887/job/31252532094?pr=1852

I'm not sure what the issue is.  I can run the notebook end to end in both Colab and Sagemaker Studio Lab without any problems other than these reported dependency issues that don't seem to affect the rest of the notebook:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jupyter-events 0.5.0 requires jsonschema[format-nongpl]>=4.3.0, but you have jsonschema 3.2.0 which is incompatible.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pydicom-seg 0.4.0 requires jsonschema<4.0.0,>=3.2.0, but you have jsonschema 4.23.0 which is incompatible.
```

thewtex commented:
Just an observation, it looks like CI is timing out (the limit is 6 hours):

![image](https://github.com/user-attachments/assets/d9688892-0a53-48cb-9aa0-e7b6d06a3c6f)

KumoLiu commented:
> Just an observation, it looks like CI is timing out (the limit is 6 hours):

Yes, but the main reason here is that it hangs at specific cell since the whole notebook don't take over 6 hours to run.

kirbyju committed:
Update TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb

Fix pep8/premerge errors.

Signed-off-by: Justin Kirby <2379527+kirbyju@users.noreply.github.com>

kirbyju commented:
> > Just an observation, it looks like CI is timing out (the limit is 6 hours):
> 
> Yes, but the main reason here is that it hangs at specific cell since the whole notebook don't take over 6 hours to run.

I just saw this in the middle of the report:

```
Running ./model_zoo/TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb
Checking PEP8 compliance...
stdin:78:1: F401 'sys' imported but unused
import sys
^
stdin:102:1: E303 too many blank lines (3)
# + [markdown] id="oYxvN-rebvwG"  # noqa: E501
^
Error: Try running with autofixes: --autofix.

Check failed!
```

I don't understand why it's complaining about sys not being used.  I guess it doesn't pick it up properly when calling it like this for all the installations?

`!{sys.executable}`

I changed it back to just say "python" instead and removed the extra blank lines.

KumoLiu commented:
Can you try fix the version to 1.0a23? From what I tested last time, I find this version will not hang. But the latest version will hang.

kirbyju committed:
Update TCIA_PROSTATEx_Prostate_MRI_Anatomy_Model.ipynb

Removing screenshots of viewers to see if that fixes pre-merge checks.

Signed-off-by: Justin Kirby <2379527+kirbyju@users.noreply.github.com>

kirbyju commented:
That was already part of the requested changes.  It's trying to use 1.0a53 in the most recent commits:

`!python -m pip install --upgrade -q \"itkwidgets[all]==1.0a53\" imjoy_elfinder`

I tried removing the image ouputs from the imjoy plugin to see if maybe that was the issue.

ericspod commented:
I attempted to run this notebook on a headless machine that we have for CICD. I found that the `view` cell just before "6. MONAI Model Inference" runs quickly but produces no view window, only messages about removing duplicate codecs. No other cell after running this will ever complete, so something changes when this is run to break the interpreter. This is with `itkwidgets[all]==1.0a53`.

I also found that `requests` needs to be explicitly installed when creating a fresh conda environment.
